
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.4.2. IMI FAIRPlus Indicators</title>
    
  <link rel="stylesheet" href="../../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/togglebutton.js"></script>
    <script src="../../../../_static/clipboard.min.js"></script>
    <script src="../../../../_static/copybutton.js"></script>
    <script src="../../../../_static/mermaid.js"></script>
    <script src="../../../../_static/mermaid.min.js"></script>
    <script src="../../../../_static/mermaid.core.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "FAIRplus/the-fair-cookbook");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "ðŸ’¬ comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/assessing-fairness/cmmi/FAIRPlus-Indicators.html" />
    <link rel="shortcut icon" href="../../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="5. FAIR assessment by hand" href="../manual-fair-assessment.html" />
    <link rel="prev" title="4.4.1. RDA Indicators" href="RDA-Indicators.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/assessing-fairness/cmmi/FAIRPlus-Indicators.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="IMI FAIRPlus Indicators" />
<meta property="og:description" content="IMI FAIRPlus Indicators  version : v0.01  FAIRplus indicators are designed for measuring data sets compliance to Data Usage Areas. One indicator might support m" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../../index.html">
  
  <img src="../../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../introduction.html">
   Introduction
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/ATI-turingway.html">
     1. The Turing Way Book of Data Science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-cookbook-audience.html">
     2. Who should read the FAIR Cookbook?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/brief-FAIR-principles.html">
     3. What are the FAIR Principles?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/what-is-IMI.html">
     4. What is IMI?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-in-IMI-context.html">
     5. FAIR in the context of IMI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/IMI-Data-Access-Request-Process.html">
     6. Engaging with IMI Projects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIRplus-values.html">
     7. The Ethical Values behind FAIR and the FAIRPlus project
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/data-sharing-models.html">
     8. High level data sharing models
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../../assessing-fairness.html">
   Assessing FAIR
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="Data-Management-Maturity-Model.html">
     1. Data Management Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Datasets-Maturity-Model.html">
     2. FAIRplus Datasets Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Data-Usage-Areas.html">
     3. Data Usage Areas
    </a>
   </li>
   <li class="toctree-l2 current active collapsible-parent">
    <a class="reference internal" href="../fair-indicators.html">
     4. FAIR Indicators
    </a>
    <ul class="current collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="RDA-Indicators.html">
       4.4.1. RDA Indicators
      </a>
     </li>
     <li class="toctree-l3 current active">
      <a class="current reference internal" href="#">
       4.4.2. IMI FAIRPlus Indicators
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../manual-fair-assessment.html">
     5. FAIR assessment by hand
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="evaluation-method.html">
       5.4.1. Evaluation Method
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="rda-evaluation.html">
       5.4.2. RDA Evaluation Methods
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../automatic-fair-assessment.html">
     6. FAIR assessment by software
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-assessment-fairevaluator.html">
       6.4.1. Assessing FAIRness with FAIREvaluator
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-assessment-fairshake.html">
       6.4.2. FAIR assessment with FAIRShake
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../infrastructure.html">
   Infrastructure for FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog.html">
     2. Building a Dataset Catalogue
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../infrastructure/vocabularies.html">
     4. Vocabularies and Terminology Infrastructure
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Technical &amp; architectural selection criteria of ontology lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selection and exploitation of Ontology Lookup Services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. How to deploy a terminology service - EBI Ontology Lookup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../findability.html">
   Improving Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers.html">
     1. How to generate globally unique, resolvable and persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/chemical-identities.html">
     2. Chemical Identity - InChI and SMILES
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers-minids.html">
     3. Identifier Minting Service with Minid Client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/zenodo-deposition.html">
     4. Deposition to Zenodo Archive
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../findability/seo.html">
     5. Findability: Search Engine Optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-data-page.html">
       5.8.1. Findability: Data Page Markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-dataset.html">
       5.8.2. Findability: Dataset Markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Findability: Data Catalog Markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../accessibility.html">
   Improving Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/anonymization.html">
     1. Anonymizing Patient Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/how-to-use-sftp-for-transfer.html">
     2. How to use SFTP to transfer data files between institutions.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/aspera.html">
     3. How to download files with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../interoperability.html">
   Improving Interoperability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/identifier-mapping.html">
     How to interlink data from different sources?
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/bridgedb-recipe.html">
     Identifier mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/bridgedb/bridgedb-example.html">
       Identifier mapping using BridgeDb WS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/selecting-ontologies.html">
     Which vocabulary to use?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/ontology-new-term-request-recipe.html">
     Requesting terms addition to terminology artefacts
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/ontology-robot-recipe.html">
     Building an application ontology with Robot
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/ontology-robot/MSIO-robot-build-process.html">
       MSIO, an application ontology for Metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/creating-minimal-metadata-profiles.html">
     Creating a Metadata Profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/transcriptomics-metadata.html">
       Establishing a Metadata Profile for Transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/covid19-sample-metadata-profile-shex-use-case.html">
       Metadata Profile Validation in RDF with Shape Expression:
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/from-proprietary-to-open-standard-mzml-exemplar.html">
     From proprietary to open standard data format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/fastq-file-format-validators.html">
     File format validation, an example with FASTQ files
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../reusability.html">
   Improving Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_software.html">
     2. Software Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_data.html">
     3. Data Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/expressing-data-use.html">
     4. Patterns and practice for expressing allowed data usage
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/provenance.html">
     5. Provenance Information
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/community-standards.html">
     6. Compliance with community annotation standards
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../applied-examples.html">
   Making IMI data FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../applied-examples/fair-data-matrix-recipe.html">
     Making a Data Matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       Bibliographic Reference:
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring a rose garden with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with ggplot2 R library
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../applied-examples/etox_raw/FAIRplus_Recipe_Ontology_mapping_the_eTox_dataset_scenario.html">
     Data normalization and integration through entity linking
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../applied-examples/nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html">
     ND4BB dataset FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     Oncotrack FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../applied-examples/resolute_raw/rawRecipe_resolute.html">
     RESOLUTE FAIRification Recipe
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../glossary-fair-terms.html">
   Glossary of FAIR things
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../help.html">
   Help
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/tips-tricks.html">
     1. Recipe Tips and Tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/recipe-template.html">
     2. Recipe Template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-gdoc.html">
     3. Using GoogleDocs to contribute a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-hackmd.html">
     4. How to use HackMD to write a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-git.html">
     5. How to create a FAIRPlus Coobook Recipe ?
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../boilerplate.html">
   Miscellaneous
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/license.html">
     1. License for this book
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/people.html">
     2. FAIR Cookbook Steering Council and Institutional Partners
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/governance.html">
     3. FAIR Cookbook Governance
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/code_of_conduct.html">
     4. FAIR Cookbook Code of Conduct
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/project_information.html">
     5. FAIR CookBook BoilerPlate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/roadmap.html">
     6. FAIR cookbook development roadmap
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/contributing.html">
     7. Contributing to the FAIR cookbook
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/disclaimer.html">
     8. Disclaimer
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../../_sources/content/recipes/assessing-fairness/cmmi/FAIRPlus-Indicators.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/assessing-fairness/cmmi/FAIRPlus-Indicators.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/assessing-fairness/cmmi/FAIRPlus-Indicators.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#index">
   4.4.2.1. Index
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#study-level">
     4.4.2.1.1. Study Level
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s01-f-s01-a-study-level-documentation-is-available-in-a-human-readable-format">
       4.4.2.1.1.1.
       <a name="F+S01">
        F+S01
       </a>
       : Study level documentation is available in a human readable format.
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s02-f-s02-a-data-is-reported-by-following-community-specific-minimum-information-guidelines">
       4.4.2.1.1.2.
       <a name="F+S02">
        F+S02
       </a>
       :
       <strong>
        Data is reported by following community specific minimum information guidelines
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s03-f-s03-a-metadata-documents-and-provides-references-about-all-data-biological-data-types-and-formats-in-data-is-expressed">
       4.4.2.1.1.3.
       <a name="F+S03">
        F+S03
       </a>
       :
       <strong>
        Metadata documents and provides references about all data biological data types and formats in data is expressed.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s04-f-s04-a-relationships-between-different-data-sets-in-a-study-is-well-defined">
       4.4.2.1.1.4.
       <a name="F+S04">
        F+S04
       </a>
       :
       <strong>
        Relationships between different data sets in a study is well defined.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s05-f-s05-a-a-versioning-policy-is-applied-to-uniquely-identify-a-particular-form-of-a-dataset-from-an-earlier-form-or-other-forms-of-itself">
       4.4.2.1.1.5.
       <a name="F+S05">
        F+S05
       </a>
       :
       <strong>
        A versioning policy is applied to uniquely identify a particular form of a dataset from an earlier form or other forms of itself.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s06-f-s06-a-share-not-only-derived-and-publication-related-data-but-data-generated-in-early-phases-of-research-data-workflow-such-as-primary-data-and-analyzed-data">
       4.4.2.1.1.6.
       <a name="F+S06">
        F+S06
       </a>
       :
       <strong>
        Share not only derived and publication related data but data generated in early phases of research data workflow such as primary data and analyzed data.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s07-f-s07-a-negative-results-are-shared">
       4.4.2.1.1.7.
       <a name="F+S07">
        F+S07
       </a>
       :
       <strong>
        Negative results are shared.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s08-f-s08-a-the-study-is-described-with-metadata-including-context-samples-and-data-acquisition-methods-for-analyzing-and-processing-data-quality-control-and-restriction-for-reuse">
       4.4.2.1.1.8.
       <a name="F+S08">
        F+S08
       </a>
       :
       <strong>
        The study is described with metadata including context, samples and data acquisition, methods for analyzing and processing data, quality control, and restriction for reuse.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s08a-f-s08a-a-metadata-includes-information-about-the-study-design-protocols-and-data-collection-methods">
       4.4.2.1.1.9.
       <a name="F+S08a">
        F+S08a
       </a>
       :
       <strong>
        Metadata includes information about the study design, protocols and data collection methods.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s08b-f-s08b-a-metadata-includes-explicit-references-to-research-resources-such-as-samples-cell-lines">
       4.4.2.1.1.10.
       <a name="F+S08b">
        F+S08b
       </a>
       :
       <strong>
        Metadata includes explicit references to research resources such as samples, cell lines
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s08c-f-s08c-a-metadata-contains-information-about-data-processing-methods-data-analysis-and-quality-assurance-metrics">
       4.4.2.1.1.11.
       <a name="F+S08c">
        F+S08c
       </a>
       :
       <strong>
        Metadata contains information about data processing methods, data analysis and quality assurance metrics.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-s08d-f-s08d-a-metadata-includes-information-about-data-ownership-license-and-reuse-constraints-for-sensitive-data">
       4.4.2.1.1.12.
       <a name="F+S08d">
        F+S08d
       </a>
       :
       <strong>
        Metadata includes information about data ownership, license and reuse constraints for sensitive data.
       </strong>
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#assays-data-set-level">
     4.4.2.1.2. Assays / Data Set Level:
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a01-f-a01-a-data-is-organized-and-documented-in-a-human-understandable-way">
       4.4.2.1.2.1.
       <a name="F+A01">
        F+A01
       </a>
       :
       <strong>
        Data is organized and documented in a human understandable way
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a02-f-a02-a-data-is-encoded-in-a-community-specific-exchange-standard">
       4.4.2.1.2.2.
       <a name="F+A02">
        F+A02
       </a>
       :
       <strong>
        Data is encoded in a community specific exchange standard.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a03-f-a03-a-a-machine-and-human-readable-formal-description-of-the-structure-of-data-is-available-including-types-properties">
       4.4.2.1.2.3.
       <a name="F+A03">
        F+A03
       </a>
       :
       <strong>
        A machine and human readable formal description of the structure of data is available including types, properties.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a04-f-a04-a-data-is-structured-by-following-a-life-sciences-domain-model-core-classes-and-their-semantic-relations-refers-to-a-common-data-model">
       4.4.2.1.2.4.
       <a name="F+A04">
        F+A04
       </a>
       :
       <strong>
        Data is structured by following a life sciences domain model, core classes and their semantic relations refers to a common data model.
       </strong>
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a05-f-a05-a-data-is-described-with-terminology-standards">
       4.4.2.1.2.5.
       <a name="F+A05">
        F+A05
       </a>
       : Data is described with terminology standards.
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#a-name-f-a06-f-a06-a-core-data-classes-important-data-elements-follows-a-common-master-and-reference-data-entity">
       4.4.2.1.2.6.
       <a name="F+A06">
        F+A06
       </a>
       :
       <strong>
        Core data classes (important data elements) follows a common master and reference data entity.
       </strong>
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   4.4.2.2. Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   4.4.2.3. License
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="imi-fairplus-indicators">
<h1><span class="section-number">4.4.2. </span>IMI FAIRPlus Indicators<a class="headerlink" href="#imi-fairplus-indicators" title="Permalink to this headline">Â¶</a></h1>
<blockquote>
<div><p>version : v0.01</p>
</div></blockquote>
<p>FAIRplus indicators are designed for measuring data sets compliance to Data Usage Areas. One indicator might support more than one Data Usage Area. Indicators are grouped according to the ISA framework.</p>
<p><img alt="ISA" src="../../../../_images/isa-structure.png" /></p>
<p>Image Reference : <a class="reference external" href="https://www.researchgate.net/publication/333163209_Design_and_Development_of_a_Phenotypic_Data_Model_PDM">Khan</a></p>
<div class="section" id="index">
<h2><span class="section-number">4.4.2.1. </span>Index<a class="headerlink" href="#index" title="Permalink to this headline">Â¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>ID</p></th>
<th class="head"><p>Indicators</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="#F+S01">F+S01</a></p></td>
<td><p>Study level documentation is available in a human readable format.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S02">F+S02</a></p></td>
<td><p>Data is reported by following community specific minimum information guidelines</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+S03">F+S03</a></p></td>
<td><p>Metadata documents and provides references about all data biological data types and formats in data is expressed.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S04">F+S04</a></p></td>
<td><p>Relationships between different data sets in a study is well defined.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+S05">F+S05</a></p></td>
<td><p>A versioning policy is applied to uniquely identify a particular form of a dataset from an earlier form or other forms of itself.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S06">F+S06</a></p></td>
<td><p>Share not only derived and publication related data but data generated in early phases of research data workflow such as primary data and analyzed data.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+S07">F+S07</a></p></td>
<td><p>Negative results are shared.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S08">F+S08</a></p></td>
<td><p>The study is described with metadata including context, samples and data acquisition, methods for analyzing and processing data, quality control, and restriction for reuse.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+S08a">F+S08a</a></p></td>
<td><p>Metadata includes information about the study design, protocols and data collection methods.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S08b">F+S08b</a></p></td>
<td><p>Metadata includes explicit references to research resources such as samples, cell lines</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+S08c">F+S08c</a></p></td>
<td><p>Metadata contains information about data processing methods, data analysis and quality assurance metrics.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+S08d">F+S08d</a></p></td>
<td><p>Metadata includes information about data ownership, license and reuse constraints for sensitive data.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+A01">F+A01</a></p></td>
<td><p>Data is organized and documented in a human understandable way</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+A02">F+A02</a></p></td>
<td><p>Data is encoded in a community specific exchange standard.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+A03">F+A03</a></p></td>
<td><p>A machine and human readable formal description of the structure of data is available including types, properties.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+A04">F+A04</a></p></td>
<td><p>Data is structured by following a life sciences domain model, core classes and their semantic relations refers to a common data model.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="#F+A05">F+A05</a></p></td>
<td><p>Data is described with terminology standards.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="#F+A06">F+A06</a></p></td>
<td><p>Core data classes (important data elements) follows a common master and reference data entity.</p></td>
</tr>
</tbody>
</table>
<div class="section" id="study-level">
<h3><span class="section-number">4.4.2.1.1. </span>Study Level<a class="headerlink" href="#study-level" title="Permalink to this headline">Â¶</a></h3>
<div class="section" id="a-name-f-s01-f-s01-a-study-level-documentation-is-available-in-a-human-readable-format">
<h4><span class="section-number">4.4.2.1.1.1. </span><a name="F+S01"> F+S01 </a> : Study level documentation is available in a human readable format.<a class="headerlink" href="#a-name-f-s01-f-s01-a-study-level-documentation-is-available-in-a-human-readable-format" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Study-level documentation provides high-level information on the research context and design, the data collection methods used, any data preparations and manipulations and summaries of findings based on the data. Examples and a suggested list of coverage can be found at <a class="reference external" href="https://www.ukdataservice.ac.uk/manage-data/document/study-level.aspx">UK Data Services</a>. <br><br>This indicator does not require any machine interpretable content, or semantic annotation of information with common vocabularies. It measures, availability of information to help data users to make informed use of the data. Publicly accessible web pages, pdf documents are acceptable formats.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s02-f-s02-a-data-is-reported-by-following-community-specific-minimum-information-guidelines">
<h4><span class="section-number">4.4.2.1.1.2. </span><a name="F+S02"> F+S02 </a> : <strong>Data is reported by following community specific minimum information guidelines</strong><a class="headerlink" href="#a-name-f-s02-f-s02-a-data-is-reported-by-following-community-specific-minimum-information-guidelines" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>A reporting standard ensures recording the information (metadata) required to unambiguously communicate experimental designs, treatments and analyses, to con-textualize the data generated. Such standards are also known as data content or minimum information standards (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/">Chervitz, et all.</a>). See examples: reporting standards for<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/table/T8/?report=objectonly"> health care</a>, for <a class="reference external" href="https://www.biorxiv.org/content/biorxiv/early/2017/07/24/167619/T4/graphic-10.large.jpg">life sciences</a> data, collection of <a class="reference external" href="https://fairsharing.org/standards/?q=minimum+information&amp;selected_facets=type_exact:reporting%20guideline">FAIRsharing </a></p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Reproducibility">Reproducibility</a> + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s03-f-s03-a-metadata-documents-and-provides-references-about-all-data-biological-data-types-and-formats-in-data-is-expressed">
<h4><span class="section-number">4.4.2.1.1.3. </span><a name="F+S03"> F+S03 </a> :  <strong>Metadata documents and provides references about all data biological data types and formats in data is expressed.</strong><a class="headerlink" href="#a-name-f-s03-f-s03-a-metadata-documents-and-provides-references-about-all-data-biological-data-types-and-formats-in-data-is-expressed" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Biological and biomedical research has been considered an especially challenging research field in this regard, as data types are extremely heterogeneous and not all have defined data standards (<a class="reference external" href="https://www.biorxiv.org/content/10.1101/167619v1.full">Griffin, et. all</a>). Metadata should capture all data types and format names in a study, if possible provide a reference or URL for format specification, if not possible have a description.<br/><br/><a class="reference external" href="https://www.biorxiv.org/highwire/markup/341342/expansion?width=1000&amp;height=500&amp;iframe=true&amp;postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed">Example </a>list of common standard data formats for omics data.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Interpretability">Interpretability</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s04-f-s04-a-relationships-between-different-data-sets-in-a-study-is-well-defined">
<h4><span class="section-number">4.4.2.1.1.4. </span><a name="F+S04"> F+S04</a> : <strong>Relationships between different data sets in a study is well defined.</strong><a class="headerlink" href="#a-name-f-s04-f-s04-a-relationships-between-different-data-sets-in-a-study-is-well-defined" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>In a study there are multiple data sets which are used as input and produced as an output. When a data is FAIRified, it is important to understand which data files are generated from the analysis of which other data sets, or sample data. For example the EMBL-EBI <a class="reference external" href="https://www.ebi.ac.uk/arrayexpress/help/creating_a_sdrf.html">SDRF </a>(Sample and Data Relationship Format) describes the sample characteristics and the relationship between samples, arrays, data files. Some communities such as proteomics have adopted these file formats to their needs, see <a class="reference external" href="https://github.com/bigbio/proteomics-metadata-standard">SDRF for Proteomics</a> . Furthermore, <a class="reference external" href="https://isa-tools.org/format/specification.html">ISA</a> allows multiomics support and is used by <a class="reference external" href="https://www.ebi.ac.uk/metabolights/index">EMBL-EBI Metabolights</a> <br><br/>The relationships can be also expressed as naming a convention. A good example is the <a class="reference external" href="https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/">TCGA </a>data set, where the barcode contains semantic information about the relation of data with other data (such as participant, sample, analyte, plate, sample). If there is a naming convention used, instead of a defined format standard, this convention should be clearly documented together with <a class="reference external" href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables">code tables</a>.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Interpretability">Interpretability</a>  + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s05-f-s05-a-a-versioning-policy-is-applied-to-uniquely-identify-a-particular-form-of-a-dataset-from-an-earlier-form-or-other-forms-of-itself">
<h4><span class="section-number">4.4.2.1.1.5. </span><a name="F+S05"> F+S05 </a> : <strong>A versioning policy is applied to uniquely identify a particular form of a dataset from an earlier form or other forms of itself.</strong><a class="headerlink" href="#a-name-f-s05-f-s05-a-a-versioning-policy-is-applied-to-uniquely-identify-a-particular-form-of-a-dataset-from-an-earlier-form-or-other-forms-of-itself" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Versioning is tracking the changes made in data by saving new copies of data files with indicators of the changes made. A new version is created when there is a change in the structure, contents, or condition of the resource. In the case of research data, a new version of a dataset may be created when an existing dataset is reprocessed, corrected or appended with additional data. Versioned data are required to cite and identify the exact dataset used as a research input in order to support research reproducibility and trustworthiness (<a class="reference external" href="https://www.ands.org.au/working-with-data/data-management/data-versioning">ANDS</a>). <br><br/>Best practices of versioning should include a numbering system, information about the status of the file, and what changes are made (see YALE Research Data Management for the Health Sciences: <a class="reference external" href="https://guides.library.yale.edu/rdm_healthsci/versioning#:~:text=Versioning">File Versioning </a> involves tracking the changes,access older copies of files.)) . PIDs can be assigned for versions, however there is no convention for that (see <a class="reference external" href="https://www.ands.org.au/working-with-data/data-management/data-versioning">ANDS</a> : DOIs for versioned data). Some simple versioning solutions can be adopted (see <a class="reference external" href="https://library.stanford.edu/research/data-management-services/data-best-practices/data-versioning">Stanford libraries</a>, Uni. <a class="reference external" href="https://data.library.virginia.edu/data-management/plan/files/">Virginia Library</a>).</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Reproducibility">Reproducibility</a> + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s06-f-s06-a-share-not-only-derived-and-publication-related-data-but-data-generated-in-early-phases-of-research-data-workflow-such-as-primary-data-and-analyzed-data">
<h4><span class="section-number">4.4.2.1.1.6. </span><a name="F+S06"> F+S06 </a> : <strong>Share not only derived and publication related data but data generated in early phases of research data workflow such as primary data and analyzed data.</strong><a class="headerlink" href="#a-name-f-s06-f-s06-a-share-not-only-derived-and-publication-related-data-but-data-generated-in-early-phases-of-research-data-workflow-such-as-primary-data-and-analyzed-data" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Life science experiments comprise different samples; based on experiment and sample, several measurements are performed. After quantification there is a step of data processing and analysis which results in life science research (<a class="reference external" href="https://www.researchgate.net/publication/51724887_A_case_study_for_efficient_management_of_high_throughput_primary_lab_data/figures">Colmsee</a> et.all). In most cases, data is shared with publications as supplementary files, or in data repositories which are referenced by articles. However raw data and primary data resides in private storages (<a class="reference external" href="https://www.researchgate.net/publication/263355191_EDAL_-_a_framework_to_store_share_and_publish_research_data/figures">Arend</a>, et. all). There are great benefits of sharing primary data, including to make meaningful comparisons between results, to answer â€˜what ifâ€™ questions and to carry out pilot studies without repeating experiments (<a class="reference external" href="https://www.nature.com/articles/nn0900_863">Koslow</a> ). <br><br>This indicator measures availability of any primary data beyond the derived data or aggregated analyzed data.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s07-f-s07-a-negative-results-are-shared">
<h4><span class="section-number">4.4.2.1.1.7. </span><a name="F+S07"> F+S07 </a> : <strong>Negative results are shared.</strong><a class="headerlink" href="#a-name-f-s07-f-s07-a-negative-results-are-shared" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Negative results are the outcomes which do not support study hypotheses. Researchers are rewarded more for publishing novel findings, and not for publishing negative results. However, sharing negative results could reduce efforts and avoid repeating work that may be difficult to replicate (<a class="reference external" href="https://www.nature.com/articles/d42473-019-00004-y#ref-CR15">ATCC</a>).Negative result also can be reported in publications (e.g. <a class="reference external" href="https://www.negative-results.org/">Negative Results Journal</a>). <br/><br>This indicator measures the availability of negative results sets in the shared data content.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s08-f-s08-a-the-study-is-described-with-metadata-including-context-samples-and-data-acquisition-methods-for-analyzing-and-processing-data-quality-control-and-restriction-for-reuse">
<h4><span class="section-number">4.4.2.1.1.8. </span><a name="F+S08"> F+S08 </a> : <strong>The study is described with metadata including context, samples and data acquisition, methods for analyzing and processing data, quality control, and restriction for reuse.</strong><a class="headerlink" href="#a-name-f-s08-f-s08-a-the-study-is-described-with-metadata-including-context-samples-and-data-acquisition-methods-for-analyzing-and-processing-data-quality-control-and-restriction-for-reuse" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>This indicator can be evaluated in two phases: 1) providing a structured metadata with domain conventions; 2) providing machine readable metadata by using any common vocabularies. <br/><br>This indicator is divided into a set of indicators from F+S08a to F+S08d.</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s08a-f-s08a-a-metadata-includes-information-about-the-study-design-protocols-and-data-collection-methods">
<h4><span class="section-number">4.4.2.1.1.9. </span><a name="F+S08a"> F+S08a </a> : <strong>Metadata includes information about the study design, protocols and data collection methods.</strong><a class="headerlink" href="#a-name-f-s08a-f-s08a-a-metadata-includes-information-about-the-study-design-protocols-and-data-collection-methods" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Example metadata: study / experiment design, trial protocol, data acquisition methods, experiment methods, data processing methods .<br/>See <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3903324/table/T1/">multi omics metadata</a> check list, <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK286004/box/box_4-1/?report=objectonly">clinical trial metadata</a></p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Reproducibility">Reproducibility</a> + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s08b-f-s08b-a-metadata-includes-explicit-references-to-research-resources-such-as-samples-cell-lines">
<h4><span class="section-number">4.4.2.1.1.10. </span><a name="F+S08b"> F+S08b</a> : <strong>Metadata includes explicit references to research resources such as samples, cell lines</strong><a class="headerlink" href="#a-name-f-s08b-f-s08b-a-metadata-includes-explicit-references-to-research-resources-such-as-samples-cell-lines" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Provide information about key research materials such as antibodies, cell lines, and organisms. Preferable with identifiers, see <a class="reference external" href="https://www.rrids.org/">RRID </a>portal.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Reproducibility">Reproducibility</a> + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s08c-f-s08c-a-metadata-contains-information-about-data-processing-methods-data-analysis-and-quality-assurance-metrics">
<h4><span class="section-number">4.4.2.1.1.11. </span><a name="F+S08c"> F+S08c</a> : <strong>Metadata contains information about data processing methods, data analysis and quality assurance metrics.</strong><a class="headerlink" href="#a-name-f-s08c-f-s08c-a-metadata-contains-information-about-data-processing-methods-data-analysis-and-quality-assurance-metrics" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>See specific examples for quality requirements for <a class="reference external" href="https://link.springer.com/chapter/10.1007/164_2019_284">In Vitro research</a>, such as chemical probes, cell line authentication, antibody validation.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Reproducibility">Reproducibility</a> + <a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-s08d-f-s08d-a-metadata-includes-information-about-data-ownership-license-and-reuse-constraints-for-sensitive-data">
<h4><span class="section-number">4.4.2.1.1.12. </span><a name="F+S08d"> F+S08d </a> :<strong>Metadata includes information about data ownership, license and reuse constraints for sensitive data.</strong><a class="headerlink" href="#a-name-f-s08d-f-s08d-a-metadata-includes-information-about-data-ownership-license-and-reuse-constraints-for-sensitive-data" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Considering the sensitive nature of life science data, beside data reuse license, consent, and if exist any other constraints should be documented.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Repurposing">Repurposing</a></p></td>
</tr>
</tbody>
</table>
<hr></hr>
</div>
</div>
<div class="section" id="assays-data-set-level">
<h3><span class="section-number">4.4.2.1.2. </span>Assays / Data Set Level:<a class="headerlink" href="#assays-data-set-level" title="Permalink to this headline">Â¶</a></h3>
<div class="section" id="a-name-f-a01-f-a01-a-data-is-organized-and-documented-in-a-human-understandable-way">
<h4><span class="section-number">4.4.2.1.2.1. </span><a name="F+A01"> F+A01 </a> : <strong>Data is organized and documented in a human understandable way</strong><a class="headerlink" href="#a-name-f-a01-f-a01-a-data-is-organized-and-documented-in-a-human-understandable-way" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Data-level, or object-level, documentation provides information at the level of variables in a database or individual objects such as images. Data-level information can be embedded in data files, such as variable, value and code labels in an SPSS file or headers in a document. Examples for quantitative, qualitative and secondary source documentation can be found at <a class="reference external" href="https://www.ukdataservice.ac.uk/manage-data/document/study-level.aspx">UK Data Services</a>.<br/><br/>This indicator does not require any machine readable content, or semantic annotation of information with common vocabularies. It measures how easily data can be explored and understood by humans who are familiar with the domain.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Interpretability">Interpretability</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-a02-f-a02-a-data-is-encoded-in-a-community-specific-exchange-standard">
<h4><span class="section-number">4.4.2.1.2.2. </span><a name="F+A02"> F+A02 </a> : <strong>Data is encoded in a community specific exchange standard.</strong><a class="headerlink" href="#a-name-f-a02-f-a02-a-data-is-encoded-in-a-community-specific-exchange-standard" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>A data exchange standard defined the encoding format of data. A data exchange standard delineates what data types can be encoded and the particular way they should be encoded (e.g., tab-delimited columns, XML, binary, etc. They facilitate the exchange of information between researchers and organizations, and between software programs or information storage systems. ). They provide syntax standards but do not specify what the document should contain in order to be considered complete (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/">Chervitz, et all.</a>).<br/></br> See examples for <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/table/T4/">omic data</a>, for microarray-based <a class="reference external" href="https://www.researchgate.net/publication/257204855_A_sea_of_standards_for_omics_data_Sink_or_swim/figures">transcriptomics </a>, for <a class="reference external" href="https://en.wikipedia.org/wiki/Clinical_Data_Interchange_Standards_Consortium#Individual_standards">clinical data</a>.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Integration">Integration</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-a03-f-a03-a-a-machine-and-human-readable-formal-description-of-the-structure-of-data-is-available-including-types-properties">
<h4><span class="section-number">4.4.2.1.2.3. </span><a name="F+A03"> F+A03 </a> : <strong>A machine and human readable formal description of the structure of data is available including types, properties.</strong><a class="headerlink" href="#a-name-f-a03-f-a03-a-a-machine-and-human-readable-formal-description-of-the-structure-of-data-is-available-including-types-properties" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>A schema describes the structure of the data. Special schemes have meanings associated with databases, such as community agreed profiles. A schema consists of a key dimension and its properties, expected types, constraints, cardinalities and associated controlled vocabularies (preferably refers to existing ontologies). Schemas and profiles can be registered and reused, for examples <a class="reference external" href="https://fairsharing.org/standards/">FAIRsharing Standards</a> and specific examples such as <a class="reference external" href="https://schema.org/docs/schemas.html">Schema.org</a>, <a class="reference external" href="https://bioschemas.org/profiles/">Bioschemas</a>, or <a class="reference external" href="https://www.hl7.org/fhir/resourcelist.html">HL7 resources</a> in the context of health data records.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Integration">Integration</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-a04-f-a04-a-data-is-structured-by-following-a-life-sciences-domain-model-core-classes-and-their-semantic-relations-refers-to-a-common-data-model">
<h4><span class="section-number">4.4.2.1.2.4. </span><a name="F+A04"> F+A04 </a> : <strong>Data is structured by following a life sciences domain model, core classes and their semantic relations refers to a common data model.</strong><a class="headerlink" href="#a-name-f-a04-f-a04-a-data-is-structured-by-following-a-life-sciences-domain-model-core-classes-and-their-semantic-relations-refers-to-a-common-data-model" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Meaningful exchange of information is a fundamental challenge in life sciences research. A domain model A domain is an abstract, implementation-independent representation of the grammar, or semantics, of a domain (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3486731/">Freimuth, et. all)</a> . Domain models define core classes and the semantic relationships between them, as well as providing unambiguous definitions for concepts required to describe life sciences research. Examples of life sciences common data models are <a class="reference external" href="https://www.ohdsi.org/data-standardization/the-common-data-model/">OMOP</a>, <a class="reference external" href="https://bridgmodel.nci.nih.gov/high-level-concept">BRIDG</a>, <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3486731/">Lifesciences DAM</a>, <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1262694/figure/F1/">functional genomics</a> data modelling.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Interpretability">Interpretability</a>  + <a class="reference external" href="Data_Usage_Areas.md#Integration">Integration</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-a05-f-a05-a-data-is-described-with-terminology-standards">
<h4><span class="section-number">4.4.2.1.2.5. </span><a name="F+A05">F+A05 </a> : Data is described with terminology standards.<a class="headerlink" href="#a-name-f-a05-f-a05-a-data-is-described-with-terminology-standards" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Terminology standards is typically defined by the use cases and provides control vocabularies to support and competency questions it is designed to answer. In life sciences domain ontologies are common ways to encode terminology standards (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/">Chervitz, et all.</a>). Terminology standards add an interpretive layer to the data by defining the concepts or terms in a domain, and in some cases the relationships between them (<a class="reference external" href="https://doi.org/10.1136/amiajnl-2013-002066">Tenenbaum et. all</a>). See an example list for <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152841/table/T5/">terminology standards</a>. For a complete listing see the <a class="reference external" href="http://www.obofoundry.org">OBO </a>Foundry.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Interpretability">Interpretability</a> + <a class="reference external" href="Data_Usage_Areas.md#Integration">Integration</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="a-name-f-a06-f-a06-a-core-data-classes-important-data-elements-follows-a-common-master-and-reference-data-entity">
<h4><span class="section-number">4.4.2.1.2.6. </span><a name="F+A06">F+A06  </a> : <strong>Core data classes (important data elements) follows a common master and reference data entity.</strong><a class="headerlink" href="#a-name-f-a06-f-a06-a-core-data-classes-important-data-elements-follows-a-common-master-and-reference-data-entity" title="Permalink to this headline">Â¶</a></h4>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Description</p></td>
<td><p>Master data is defined as core business objects used in different applications across an organization along with their associated metadata, definitions and taxonomies. Reference data is used to characterize or classify other data such as codes and description tables. Master and Reference data lowers cost and complexity through use of standards, common data models, and integration patterns. Sharing master data within a community or in organization reduces variability caused by multiple studies producing the same type of data, but in isolation, then inconsistencies in data structure and data values between the systems occurs (DAMA). <br>In the life science domain, master data entities examples can be a study, a subject, a file, a chemical compound, or an observation. Common data models provide core classes which can be useful for creating master data elements (see<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3486731/"> LS DAM </a>Experiment, Molecular Biology, Molecular Databases core classes). However master data should be described, are assigned an unique identifier, and registered on a searchable source. An example to reference data from the health domain can be the <a class="reference external" href="https://simplifier.net/guide/ProfilingAcademy/Terminology">value set</a> of the HL7.<br><br>Precondition of measuring this indicator is the existence of a master and reference data in a related research community, an organization, or a repository.</p></td>
</tr>
<tr class="row-odd"><td><p>Related DU Area</p></td>
<td><p><a class="reference external" href="Data_Usage_Areas.md#Integration">Integration</a></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="authors">
<h2><span class="section-number">4.4.2.2. </span>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">Â¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Name</p></th>
<th class="text-align:left head"><p>Affiliation</p></th>
<th class="text-align:left head"><p>orcid</p></th>
<th class="text-align:left head"><p>CrediT role</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>Oya Deniz Beyan</p></td>
<td class="text-align:left"><p><a class="reference external" href="http://fit.fraunhofer.de">fit.fraunhofer.de</a></p></td>
<td class="text-align:left"><p><a class="reference external" href="https://orcid.org/orcid.org/0000-0000-0000-0000">0000-0000-0000-0000</a></p></td>
<td class="text-align:left"><p>Writing - Original Draft</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>Writing - Review &amp; Editing, Funding acquisition</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="license">
<h2><span class="section-number">4.4.2.3. </span>License<a class="headerlink" href="#license" title="Permalink to this headline">Â¶</a></h2>
<p>This page is released under the Creative Commons 4.0 BY license.</p>
<p><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png" height="20"/></a></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/assessing-fairness/cmmi"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="RDA-Indicators.html" title="previous page"><span class="section-number">4.4.1. </span>RDA Indicators</a>
    <a class='right-next' id="next-link" href="../manual-fair-assessment.html" title="next page"><span class="section-number">5. </span>FAIR assessment by hand</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The FAIRplus project members<br/>
        
            &copy; Copyright 2020 (CC-BY 4.0).<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="row_footer"> This project has received funding from the <a href="https://www.imi.europa.eu/">Innovative Medicines Initiative</a> 2 Joint Undertaking under grant agreement No 802750. This Joint Undertaking receives support from the European Unionâ€™s Horizon 2020 research and innovation programme and EFPIA Companies. This communication reflects the views of the authors and neither IMI nor the European Union, EFPIA or any Associated Partners are liable for any use that may be made of the information contained herein. </div>
<div class="row_footer"> This page was created by <a href="https://github.com/jupyter/jupyter-book/graphs/contributors">The Jupyter Book Community</a> </div>
<div class="row_footer"> <div class="column" style="center"><a href="https://europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/eu-flag-recolored-fairplus.png" height="20" alt="EU flag" style="float:center""></a></div> <div class="column"><a href="https://www.imi.europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/imi-logo-fairplus-v1.png" height="20" alt="IMI logo" style="float:center""></a></div> <div class="column"><a href="https://efpia.eu"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/efpia-logo-fairplus2.png" height="20" alt="EFPIA logo" style="float:center""></a></div> <br/> </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>