
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Identifier mapping with BridgeDB</title>
    
  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/mermaid.js"></script>
    <script src="../../../_static/mermaid.min.js"></script>
    <script src="../../../_static/mermaid.core.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "FAIRplus/the-fair-cookbook");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "💬 comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/bridgedb-recipe.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Identifier mapping using BridgeDb WS" href="bridgedb/bridgedb-example.html" />
    <link rel="prev" title="How to interlink data from different sources?" href="identifier-mapping.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/bridgedb-recipe.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Identifier mapping with BridgeDB" />
<meta property="og:description" content="Identifier mapping with BridgeDB    &lt;br/&gt;     &lt;i class=&#34;fa fa-qrcode fa-2x&#34; style=&#34;color:#7e0038;&#34;&gt;&lt;/i&gt;  &lt;h4&gt;&lt;b&gt;Recipe metadata&lt;/b&gt;&lt;/h4&gt;  identifier: &lt;a href=&#34;&#34;" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../index.html">
  
  <img src="../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../introduction.html">
   Introduction
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/ATI-turingway.html">
     1. The Turing Way Book of Data Science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
     2. Who should read the FAIR Cookbook?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
     3. What are the FAIR Principles?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/what-is-IMI.html">
     4. What is IMI?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIR-in-IMI-context.html">
     5. FAIR in the context of IMI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/IMI-Data-Access-Request-Process.html">
     6. Engaging with IMI Projects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIRplus-values.html">
     7. The Ethical Values behind FAIR and the FAIRPlus project
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/data-sharing-models.html">
     8. High level data sharing models
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessing FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Data-Management-Maturity-Model.html">
     1. Data Management Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Datasets-Maturity-Model.html">
     2. FAIRplus Datasets Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Data-Usage-Areas.html">
     3. Data Usage Areas
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/fair-indicators.html">
     4. FAIR Indicators
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/RDA-Indicators.html">
       4.4.1. RDA Indicators
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/FAIRPlus-Indicators.html">
       4.4.2. IMI FAIRPlus Indicators
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/manual-fair-assessment.html">
     5. FAIR assessment by hand
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/evaluation-method.html">
       5.4.1. Evaluation Method
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/rda-evaluation.html">
       5.4.2. RDA Evaluation Methods
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/automatic-fair-assessment.html">
     6. FAIR assessment by software
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
       6.4.1. Assessing FAIRness with FAIREvaluator
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
       6.4.2. FAIR assessment with FAIRShake
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure for FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     2. Building a Dataset Catalogue
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     4. Vocabularies and Terminology Infrastructure
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Technical &amp; architectural selection criteria of ontology lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selection and exploitation of Ontology Lookup Services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. How to deploy a terminology service - EBI Ontology Lookup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Improving Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. How to generate globally unique, resolvable and persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Chemical Identity - InChi and SMILES
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     3. Identifier Minting Service with Minid Client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     4. Deposition to Zenodo Archive
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     5. Findability: Search Engine Optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       5.8.1. Findability: Data Page Markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       5.8.2. Findability: Dataset Markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Findability: Data Catalog Markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Improving Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/anonymization.html">
     1. Anonymizing Patient Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/how-to-use-sftp-for-transfer.html">
     2. How to use SFTP to transfer data files between institutions.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     3. How to download files with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Improving Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     How to interlink data from different sources?
    </a>
   </li>
   <li class="toctree-l2 current active collapsible-parent">
    <a class="current reference internal" href="#">
     Identifier mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       Identifier mapping using BridgeDb WS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     Which vocabulary to use?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     Requesting terms addition to terminology artefacts
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="ontology-robot-recipe.html">
     Building an application ontology with Robot
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot/MSIO-robot-build-process.html">
       MSIO, an application ontology for Metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     Creating a Metadata Profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="transcriptomics-metadata.html">
       Establishing a Metadata Profile for Transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       Metadata Profile Validation in RDF with Shape Expression:
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="from-proprietary-to-open-standard-mzml-exemplar.html">
     From proprietary to open standard data format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     File format validation, an example with FASTQ files
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Improving Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Software Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Data Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Patterns and practice for expressing allowed data usage
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Provenance Information
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/community-standards.html">
     6. Compliance with community annotation standards
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Making IMI data FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making a Data Matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       Bibliographic Reference:
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring a rose garden with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with ggplot2 R library
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox_raw/FAIRplus_Recipe_Ontology_mapping_the_eTox_dataset_scenario.html">
     Data normalization and integration through entity linking
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html">
     ND4BB dataset FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     Oncotrack FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/rawRecipe_resolute.html">
     RESOLUTE FAIRification Recipe
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../glossary-fair-terms.html">
   Glossary of FAIR things
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Help
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     1. Recipe Tips and Tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     2. Recipe Template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     3. Using GoogleDocs to contribute a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     4. How to use HackMD to write a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     5. How to create a FAIRPlus Coobook Recipe ?
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../boilerplate.html">
   Miscellaneous
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/license.html">
     1. License for this book
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/people.html">
     2. FAIR Cookbook Steering Council and Institutional Partners
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/governance.html">
     3. FAIR Cookbook Governance
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/code_of_conduct.html">
     4. FAIR Cookbook Code of Conduct
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/project_information.html">
     5. FAIR CookBook BoilerPlate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/roadmap.html">
     6. FAIR cookbook development roadmap
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/contributing.html">
     7. Contributing to the FAIR cookbook
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/disclaimer.html">
     8. Disclaimer
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../_sources/content/recipes/interoperability/bridgedb-recipe.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/bridgedb-recipe.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/interoperability/bridgedb-recipe.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   Main Objectives
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#graphical-overview">
   Graphical Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#requirements">
   Requirements
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#capability-maturity-table">
   Capability &amp; Maturity Table
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#tools">
   Tools
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#id1">
   Identifier mapping with BridgeDb
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cases">
     Cases:
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-global-identifier-to-other-global-identifier">
     Mapping global identifier to other global identifier
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#bridgedb-via-webservices-using-python">
       BridgeDb via Webservices using Python
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#bridgedb-via-the-dedicated-r-package">
       BridgeDb via the dedicated R package
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-local-identifier-to-a-different-global-identifier">
     Mapping local identifier to a different global identifier
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#webservices-in-python">
       Webservices in Python
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#r-package">
       R Package
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#provenance">
   Provenance
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#code">
   Code
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   Conclusion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-should-i-read-next">
     What should I read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   License
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="fcb-bridgedb">
<span id="identifier-mapping-with-bridgedb"></span><h1>Identifier mapping with BridgeDB<a class="headerlink" href="#fcb-bridgedb" title="Permalink to this headline">¶</a></h1>
<br/>
<hr class="docutils" />
<div class="sphinx-bs container-lg pb-3 docutils">
<div class="row docutils">
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-qrcode fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Recipe metadata</b></h4>
 identifier: <a href="">RX.X</a> 
 version: <a href="">v1.0</a>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-fire fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Difficulty level</b></h4>
<i class="fa fa-fire fa-lg" style="color:#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:lightgrey"></i>
<i class="fa fa-fire fa-lg" style="color:lightgrey"></i>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fas fa-clock fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Reading Time</b></h4>
<i class="fa fa-clock fa-lg" style="color:#7e0038;"></i> 30 minutes
<h4><b>Recipe Type</b></h4>
<i class="fa fa-globe fa-lg" style="color:#7e0038;"></i> Hands-on
<h4><b>Executable Code</b></h4>
<i class="fa fa-play-circle fa-lg" style="color:#7e0038;"></i> Yes
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-users fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Intended Audience</b></h4>
<p> <i class="fa fa-user-md fa-lg" style="color:#7e0038;"></i> Principal Investigators </p>
<p> <i class="fa fa-database fa-lg" style="color:#7e0038;"></i> Data Manager </p>
<p> <i class="fa fa-wrench fa-lg" style="color:#7e0038;"></i> Data Scientist </p>
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="main-objectives">
<h2>Main Objectives<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¶</a></h2>
<p>The main purpose of this recipe is to:</p>
<blockquote>
<div><p>Provide practical examples on how to map <strong>identifiers for molecular entities</strong> (<code class="docutils literal notranslate"><span class="pre">such</span> <span class="pre">as</span> <span class="pre">Genes,</span> <span class="pre">Proteins,</span> <span class="pre">Metabolites</span> <span class="pre">and</span> <span class="pre">Pathways</span></code>) between resources using a purpose built tool, namely <a class="reference external" href="https://bio.tools/bridgedb">BridgeDB</a>.
Hands on guidance is provided for 2 interfaces (R package and a Python Webservices) provided by <a class="reference external" href="https://bio.tools/bridgedb">BridgeDB</a>.</p>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="graphical-overview">
<h2>Graphical Overview<a class="headerlink" href="#graphical-overview" title="Permalink to this headline">¶</a></h2>
<p>This recipe will cover the highlighted topics</p>
<div class="figure align-default" id="overview-of-key-aspects-in-identifier-mapping">
<a class="reference internal image-reference" href="../../../_images/id-map-service-mermaid.png"><img alt="Overview of key aspects in  Identifier Mapping" src="../../../_images/id-map-service-mermaid.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2 </span><span class="caption-text">Overview of key aspects in  Identifier Mapping</span><a class="headerlink" href="#overview-of-key-aspects-in-identifier-mapping" title="Permalink to this image">¶</a></p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>recipe dependency:</p>
<ul>
<li><p><span class="xref std std-ref">fcb-identifiers</span></p></li>
<li><p><span class="xref std std-ref">fcb-idmapping</span></p></li>
</ul>
</li>
<li><p>skill dependency:</p>
<ul>
<li><p>programming knowledge</p></li>
<li><p>R</p></li>
<li><p>Python</p></li>
</ul>
</li>
<li><p>technical dependency</p>
<ul>
<li><p>R or Python environment</p></li>
<li><p>BridgeDB R package installed</p></li>
</ul>
</li>
</ul>
</div>
<hr class="docutils" />
<div class="section" id="capability-maturity-table">
<h2>Capability &amp; Maturity Table<a class="headerlink" href="#capability-maturity-table" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Capability</p></th>
<th class="text-align:left head"><p>Initial Maturity Level</p></th>
<th class="text-align:left head"><p>Final Maturity Level</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>Interoperability</p></td>
<td class="text-align:left"><p>minimal</p></td>
<td class="text-align:left"><p>repeatable</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Reusability</p></td>
<td class="text-align:left"><p>minimal</p></td>
<td class="text-align:left"><p>reusable</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="tools">
<h2>Tools<a class="headerlink" href="#tools" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Software</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>version</p></th>
<th class="head"><p>Biotools record</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://bridgedb.github.io/">BridgeDb webservices</a></p></td>
<td><p>BridgeDb is a framework to map identifiers between various databases. It includes a Java library that provides an API to work with identifier-identifier mapping databases and resources.</p></td>
<td><p>0.0.9</p></td>
<td><p><a class="reference external" href="https://bio.tools/bridgedb">https://bio.tools/bridgedb</a></p></td>
</tr>
<tr class="row-odd"><td><p>Python</p></td>
<td><p>an interpreted, high-level and general-purpose programming language</p></td>
<td><p>3.8.5</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://pandas.pydata.org/">Pandas</a></p></td>
<td><p>pandas is a fast, powerful, flexible and easy to use open source data analysis and manipulation tool,built on top of the Python programming language.</p></td>
<td><p>1.1.3</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>R</p></td>
<td><p>R is a programming language and free software environment for statistical computing and graphics supported by the R Foundation for Statistical Computing.</p></td>
<td><p>v4.0.3</p></td>
<td><p><a class="reference external" href="https://bio.tools/r">https://bio.tools/r</a></p></td>
</tr>
<tr class="row-even"><td><p>tidyverse</p></td>
<td><p>The tidyverse is an opinionated collection of R packages designed for data science.</p></td>
<td><p>1.3.0</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>BridgeDbR</p></td>
<td><p>an R package for BridgeDb</p></td>
<td><p>2.0.0</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="id1">
<h2>Identifier mapping with BridgeDb<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p><span class="xref std std-ref">fcb-idmapping</span> is an essential step for data reusability and interoperability. This step requires dedicated tools. With the present recipe, we show how to use BridgeDb to carry out  this process.</p>
<p>BridgeDB is an open source tool dedicated to performing identifier mapping. BridgeDB offers three different interfaces:</p>
<ul class="simple">
<li><p>Java API</p></li>
<li><p>R package</p></li>
<li><p>REST Web-services</p></li>
</ul>
<blockquote>
<div><p>:book: In the context of this recipe, we distinguish between two types of identifiers:</p>
<ul class="simple">
<li><p><em>Local identifiers</em>, which refer to identifiers that are minted within an organization or database and thus internally defined (i.e. local to said organization)</p></li>
<li><p><em>Global identifiers</em>, which refer to identifiers which globally unique and uniquely point to an entity, as available from BridgeDb’s <a class="reference external" href="https://github.com/bridgedb/BridgeDb/blob/2dba5780260421de311cb3064df79e16a396b887/org.bridgedb.bio/resources/org/bridgedb/bio/datasources.tsv">data sources file</a></p></li>
</ul>
</div></blockquote>
<p>We will focus here on two distinct cases, depending on the nature of the incoming data. Namely, whether our data is already using global identifiers or only relies on local identifiers.</p>
<div class="section" id="cases">
<h3>Cases:<a class="headerlink" href="#cases" title="Permalink to this headline">¶</a></h3>
<ol class="simple">
<li><p>Connect data with global identifiers to different global identifiers</p></li>
<li><p>Connect data with local identifiers that were mapped to a global identifier to a different global identifier</p></li>
</ol>
<p>In this recipe, we will cover how the R package and webservices can be used to accomplish the stated objectives.</p>
</div>
<div class="section" id="mapping-global-identifier-to-other-global-identifier">
<h3>Mapping global identifier to other global identifier<a class="headerlink" href="#mapping-global-identifier-to-other-global-identifier" title="Permalink to this headline">¶</a></h3>
<p>In this case, the input data is a list of elements with an identifier that is part of <a class="reference external" href="https://github.com/bridgedb/BridgeDb/blob/2dba5780260421de311cb3064df79e16a396b887/org.bridgedb.bio/resources/org/bridgedb/bio/datasources.tsv">BridgeDb’s data sources</a>. In our example, we will use a list of Homo Sapiens, Hugo Gene Nomenclature Convention (<a class="reference external" href="http://www.genenames.org">HGNC</a>) gene identifiers stored in a TSV file. The objective is to map these to other available gene identifiers.</p>
<div class="section" id="bridgedb-via-webservices-using-python">
<h4>BridgeDb via Webservices using Python<a class="headerlink" href="#bridgedb-via-webservices-using-python" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><p>:exclamation: For this tutorial Python v3.8.5, <a class="reference external" href="https://pandas.pydata.org/">pandas</a> v1.1.3 and BridgeDb Webservices v0.9.0 were used.</p>
</div></blockquote>
<p>One of the biggest benefits of using BridgeDb webservices is that these can be accessed using most programming language. Python has become one of the leading programming languages in data science and predictive modelling. Despite the lack of a dedicated BridgeDb Python library, we show here how to use the BridgeDb Webservices to perform the mappings suggested under <a class="reference external" href="#Cases">Cases</a></p>
<ul class="simple">
<li><p>We start by defining strings containing the url of the webservices and the specific method from the Webservices we want to use. In our case, a <code class="docutils literal notranslate"><span class="pre">batch</span> <span class="pre">cross</span> <span class="pre">reference</span></code>. When doing the query, we need to specify <strong>the organism</strong> and <strong>the source dataset</strong>. We can also <em>optionally</em> specify a <em>target data source</em> if we only want to map one of them (e.g. Ensembl)</p></li>
</ul>
<div class="highlight-python=3.8.5 notranslate"><div class="highlight"><pre><span></span>url = &quot;https://webservice.bridgedb.org/&quot;
batch_request = url+&quot;{org}/xrefsBatch/{source}{}&quot;
</pre></div>
</div>
<blockquote>
<div><p>If the aim is to only map to a specific target data source, then one can check whether the mapping is supported by invoking the following  webservices:</p>
</div></blockquote>
<div class="highlight-python=3.8.5 notranslate"><div class="highlight"><pre><span></span>mapping_available = &quot;{org}/isMappingSupported/{source}/{target}&quot;
query = url+mapping_available.format(org=&#39;Homo sapiens&#39;, source=&#39;H&#39;, target=&#39;En&#39;)
requests.get(query).text
</pre></div>
</div>
<blockquote>
<div><p>This will return <code class="docutils literal notranslate"><span class="pre">True</span></code> if the mapping between the given source and target is supported for the given organism or <code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</div></blockquote>
<ul class="simple">
<li><p>We then load our data into a pandas dataframe and call the requests library using our query.</p></li>
</ul>
<div class="highlight-python=3.8.5 notranslate"><div class="highlight"><pre><span></span>query = batch_request.format(&#39;?dataSource=En&#39;, org=&#39;Homo sapiens&#39;, source=&#39;H&#39;)
response = requests.post(query, data=data.to_csv(index=False, header=False))
</pre></div>
</div>
<blockquote>
<div><p>The webservice response is now stored in the <code class="docutils literal notranslate"><span class="pre">response</span></code> variable. We can then simply pass this variable to the <code class="docutils literal notranslate"><span class="pre">to_df</span></code> method provided in the <code class="docutils literal notranslate"><span class="pre">bridgedb_script.py</span></code> module (see <a class="reference external" href="#Code">Code</a>). This method will extract the response in text form and turn it into a pandas Dataframe with conveniently named columns and structured data.</p>
</div></blockquote>
<p>The output table will contain:</p>
<ul class="simple">
<li><p>The original identifier</p></li>
<li><p>The data source that the identifier is part of</p></li>
<li><p>The mapped identifier</p></li>
<li><p>The data source for the mapped identifier</p></li>
</ul>
<blockquote>
<div><p>In our case the output of <code class="docutils literal notranslate"><span class="pre">to_df</span></code> is:</p>
</div></blockquote>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>original</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>mapping</p></th>
<th class="text-align:left head"><p>target</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000121410</p></td>
<td class="text-align:left"><p>En</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1CF</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000148584</p></td>
<td class="text-align:left"><p>En</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A2MP1</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000256069</p></td>
<td class="text-align:left"><p>En</p></td>
</tr>
</tbody>
</table>
<blockquote>
<div><p>If we were to not specify the target data source (by passing an empty string as the parameter), we would get all the potential mappings for the given identifiers. In our case (top 10 rows):</p>
</div></blockquote>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>original</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>mapping</p></th>
<th class="text-align:left head"><p>target</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc002qsd.5</p></td>
<td class="text-align:left"><p>Uc</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>8039748</p></td>
<td class="text-align:left"><p>X</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0072562</p></td>
<td class="text-align:left"><p>T</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc061drj.1</p></td>
<td class="text-align:left"><p>Uc</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ILMN_2055271</p></td>
<td class="text-align:left"><p>Il</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>Hs.529161</p></td>
<td class="text-align:left"><p>U</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0070062</p></td>
<td class="text-align:left"><p>T</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0002576</p></td>
<td class="text-align:left"><p>T</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc061drt.1</p></td>
<td class="text-align:left"><p>Uc</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>51020_at</p></td>
<td class="text-align:left"><p>X</p></td>
</tr>
</tbody>
</table>
<p>As one can see, using BridgeDb service via Python is extremely simple and can be easily integrated in an annotation pipeline.</p>
</div>
<div class="section" id="bridgedb-via-the-dedicated-r-package">
<h4>BridgeDb via the dedicated R package<a class="headerlink" href="#bridgedb-via-the-dedicated-r-package" title="Permalink to this headline">¶</a></h4>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For this tutorial R v4.0.3, <a class="reference external" href="https://www.tidyverse.org/">tidyverse</a> v1.3.0 and <a class="reference external" href="https://www.bioconductor.org/packages/release/bioc/html/BridgeDbR.html">BridgeDbR</a> v2.0.0 were used.</p>
</div>
<p>After having loaded the required R libraries, we read the data and create a new column to include the source of the identifier.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">data_df</span> <span class="o">&lt;-</span> <span class="nf">read_tsv</span><span class="p">(</span><span class="n">filepath</span><span class="p">,</span> <span class="n">col_names</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;identifier&#39;</span><span class="p">))</span>
<span class="n">data_df</span><span class="o">$</span><span class="n">source</span> <span class="o">=</span> <span class="s">&#39;H&#39;</span>
</pre></div>
</div>
<p>We then load the data for the organism we are mapping from.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span> <span class="n">location</span> <span class="o">&lt;-</span> <span class="nf">getDatabase</span><span class="p">(</span><span class="s">&#39;Homo sapiens&#39;</span><span class="p">)</span>
 <span class="n">mapper</span> <span class="o">&lt;-</span> <span class="nf">loadDatabase</span><span class="p">(</span><span class="n">location</span><span class="p">)</span>
</pre></div>
</div>
<p>And use the library’s     dedicated function to map the identifiers:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">maps</span><span class="p">(</span><span class="n">mapper</span><span class="p">,</span> <span class="n">data_df</span><span class="p">,</span> <span class="n">target</span><span class="o">=</span><span class="s">&#39;En&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>This will return:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>identifier</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>mapping</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000121410</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1CF</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000148584</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A2MP1</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000256069</p></td>
</tr>
</tbody>
</table>
<p>As seen earlier when using Python language, we can obtain all possible mappings simply by not specifying the target. This will result in (top 10)</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>identifier</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>mapping</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc002qsd.5</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>8039748</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0072562</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc061drj.1</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Il</p></td>
<td class="text-align:left"><p>ILMN_2055271</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>U</p></td>
<td class="text-align:left"><p>Hs.529161</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0070062</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0002576</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc061drt.1</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>51020_at</p></td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>An error message indicating “Error in download.file” may be thrown. If the case, it may be caused by <code class="docutils literal notranslate"><span class="pre">timeout</span></code> variable being set to too small a value. To remediate the issue, try increasing the timeout variable value by calling <code class="docutils literal notranslate"><span class="pre">options(timeout=300)</span></code></p>
</div>
<!--
This will maybe link to the identifier mapping recipe in the future, where there will be a specific section detailing identifier equivalence files taking into account scientific lenses, as discussed with Egon. 

--------------------------------


### Mapping local identifier to global identifier
[NEED HELP HERE]
This is a step that should be done manually. In this case an important decision should be made, which identifiers to map to. To decide this you should look for the data source that contains the closest definition of the concepts to the one you used for your local identifiers.

-->
</div>
</div>
<div class="section" id="mapping-local-identifier-to-a-different-global-identifier">
<h3>Mapping local identifier to a different global identifier<a class="headerlink" href="#mapping-local-identifier-to-a-different-global-identifier" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In this section, we assume that we already have an equivalence file containing the mapping of a local identifier to one of the global identifiers. In our case, this will be contained in a TSV where we map our local gene identifier to <a class="reference external" href="http://www.genenames.org">HGNC</a>. One may consult the list of other potential data formats in the <span class="xref std std-ref">fcb-idmapping</span> recipe. The mapping should be <strong>one-to-one</strong> for this recipe.</p>
</div>
<p>The TSV mapping file looks as follows:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>local</p></th>
<th class="text-align:left head"><p>source</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>bb34</p></td>
<td class="text-align:left"><p>A1CF</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>eg93</p></td>
<td class="text-align:left"><p>A2MP1</p></td>
</tr>
</tbody>
</table>
<p>You may notice the <code class="docutils literal notranslate"><span class="pre">source</span></code> identifiers correspond with those used in the previous example.</p>
<p>This is how the mapping will work</p>
<div class="figure align-default" id="overview-of-bridgedb-tools">
<a class="reference internal image-reference" href="../../../_images/bridgedb-fig1-mermaid.png"><img alt="Overview of BridgeDB tools" src="../../../_images/bridgedb-fig1-mermaid.png" style="width: 500px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3 </span><span class="caption-text">Overview of BridgeDB tools</span><a class="headerlink" href="#overview-of-bridgedb-tools" title="Permalink to this image">¶</a></p>
</div>
<div class="section" id="webservices-in-python">
<h4>Webservices in Python<a class="headerlink" href="#webservices-in-python" title="Permalink to this headline">¶</a></h4>
<p>As before, we will define variables including the <code class="docutils literal notranslate"><span class="pre">web-service's</span> <span class="pre">URL</span></code> and the <code class="docutils literal notranslate"><span class="pre">method</span></code> that we will use, in this instance: <code class="docutils literal notranslate"><span class="pre">xRefsBatch</span></code>.</p>
<p>We then pass the source column to the <code class="docutils literal notranslate"><span class="pre">post</span> <span class="pre">request</span></code> as follows</p>
<div class="highlight-python=3.8.5 notranslate"><div class="highlight"><pre><span></span>source_data = case2.source.to_csv(index=False, header=False)
query = batch_request.format(&#39;&#39;, org=org, source=source)
response2 = requests.post(query, data = source_data)
</pre></div>
</div>
<p>You may notice here that we did not pass a target source, this could be done as specified before. Then, we use <code class="docutils literal notranslate"><span class="pre">to_df</span></code> again and as expected obtain the same dataframe as before.
To see the equivalences with our local identifiers, we can simply join the dataframes, as follows:</p>
<div class="highlight-python=3.8.5 notranslate"><div class="highlight"><pre><span></span>local_mapping = mappings.join(case2.set_index(&#39;source&#39;), on=&#39;original&#39;)
</pre></div>
</div>
<p>which will return the following table (first 10 rows)</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>original</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>mapping</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>local</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc002qsd.5</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>8039748</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0072562</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc061drj.1</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ILMN_2055271</p></td>
<td class="text-align:left"><p>Il</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>Hs.529161</p></td>
<td class="text-align:left"><p>U</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0070062</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>GO:0002576</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>uc061drt.1</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>51020_at</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
</tbody>
</table>
<p>In case we did specify the <code class="docutils literal notranslate"><span class="pre">target</span></code> argument to be <code class="docutils literal notranslate"><span class="pre">Ensembl</span> <span class="pre">(En)</span></code>, we would instead get</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>original</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>mapping</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>local</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000121410</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>aa11</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>A1CF</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000148584</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>bb34</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>A2MP1</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>ENSG00000256069</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>eg93</p></td>
</tr>
</tbody>
</table>
<p>Here, we see a <code class="docutils literal notranslate"><span class="pre">one-to-one</span></code> relation between the identifiers in HGNC and En while the relation between HGNC and UCSC Genome Browser (Uc) or Gene Ontology (T) is <code class="docutils literal notranslate"><span class="pre">one-to-many</span></code>. Depending on the identifiers and resources, the relation could also be <code class="docutils literal notranslate"><span class="pre">many-to-many</span></code> as shown below.</p>
<!--
You may notice that despite the 1-to-1 relation between `local` and `original` we get a N-to-N relation between `local` and `mapping` due to the N-to-N relation between `original` and `mapping`. This can be easily understood with the diagram below-->
<div class="figure align-default" id="id2">
<a class="reference internal image-reference" href="../../../_images/bridgedb-fig2-mermaid.png"><img alt="Overview of BridgeDB tools" src="../../../_images/bridgedb-fig2-mermaid.png" style="width: 500px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4 </span><span class="caption-text">Overview of BridgeDB tools</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This many-to-many relationship stems from different <em>scientific lenses</em> in the data sources. You can read more about these in <span id="id3">[<span>batchelor_scientific_nodate</span>]</span>. The core idea is that depending on the domain/application of the data we can consider different entities as unique. While certain proteins could be considered “equal” from a biological perspective they may require differentiation when using a chemical lense. This is what then leads to many-to-many relationships.</p>
</div>
</div>
<div class="section" id="r-package">
<h4>R Package<a class="headerlink" href="#r-package" title="Permalink to this headline">¶</a></h4>
<p>Here, we will follow the same steps as in the previous case. The only difference is that we need to specify the columns/fields to use when loading the data:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">data_df</span> <span class="o">&lt;-</span> <span class="nf">read_tsv</span><span class="p">(</span><span class="n">filepath</span><span class="p">,</span> <span class="n">col_names</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;local&#39;</span><span class="p">,</span> <span class="s">&#39;identifier&#39;</span><span class="p">))</span>
</pre></div>
</div>
<p>Then, after computing the mapping, we can join it with the local identifier</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">right_join</span><span class="p">(</span><span class="n">data_df</span><span class="p">,</span> <span class="n">mapping</span><span class="p">)</span>
</pre></div>
</div>
<p>Assuming we did not specify the target data source we obtain the following table (first 10 rows):</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>local</p></th>
<th class="text-align:left head"><p>identifier</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>mapping</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc002qsd.5</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>8039748</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0072562</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc061drj.1</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Il</p></td>
<td class="text-align:left"><p>ILMN_2055271</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>U</p></td>
<td class="text-align:left"><p>Hs.529161</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0070062</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>T</p></td>
<td class="text-align:left"><p>GO:0002576</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>Uc</p></td>
<td class="text-align:left"><p>uc061drt.1</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>H</p></td>
<td class="text-align:left"><p>X</p></td>
<td class="text-align:left"><p>51020_at</p></td>
</tr>
</tbody>
</table>
<p>In case we did specify the target data source we would get:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>local</p></th>
<th class="text-align:left head"><p>identifier</p></th>
<th class="text-align:left head"><p>source</p></th>
<th class="text-align:left head"><p>target</p></th>
<th class="text-align:left head"><p>mapping</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>aa11</p></td>
<td class="text-align:left"><p>A1BG</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000121410</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>bb34</p></td>
<td class="text-align:left"><p>A1CF</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000148584</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>eg93</p></td>
<td class="text-align:left"><p>A2MP1</p></td>
<td class="text-align:left"><p>HGNC</p></td>
<td class="text-align:left"><p>En</p></td>
<td class="text-align:left"><p>ENSG00000256069</p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="provenance">
<h2>Provenance<a class="headerlink" href="#provenance" title="Permalink to this headline">¶</a></h2>
<p>BridgeDb provides provenance information through:</p>
<ul class="simple">
<li><p>A call to /properties/ in the Webservices</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">getProperties()</span></code> in BridgeDbR (passing the mapper as a parameter)</p></li>
</ul>
<p>This returns the following information for each of the data sources for a given organism:</p>
<ul class="simple">
<li><p>Data source name</p></li>
<li><p>Build date</p></li>
<li><p>Series</p></li>
<li><p>Data type</p></li>
<li><p>Data source version</p></li>
<li><p>Schema version</p></li>
</ul>
<p>Improvements on provenance are under way (see <a class="reference external" href="https://github.com/bridgedb/BridgeDb/issues/164">here</a>)</p>
</div>
<hr class="docutils" />
<div class="section" id="code">
<h2>Code<a class="headerlink" href="#code" title="Permalink to this headline">¶</a></h2>
<p>You can find ready-made methods to map using R and Python for the given use cases <a class="reference external" href="https://github.com/FAIRplus/the-fair-cookbook/tree/bridgedb/docs/content/recipes/interoperability/bridgedb">here</a>. These assume the data has the structure described in this recipe.</p>
</div>
<hr class="docutils" />
<div class="section" id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p>We showed how to use BridgeDb webservices and R package to map identifiers from different data sources using a minimal dataset.
BridgeDb provides handy functionality to make ‘omics’ type of data more interoperable and reusable.
As with all annotation services, it is important to bear in mind the version of the service being used as well as the data on which the service invokation has been performed.
These are aspects of information provenance which</p>
</div></blockquote>
<div class="section" id="what-should-i-read-next">
<h3>What should I read next?<a class="headerlink" href="#what-should-i-read-next" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../findability/identifiers.html#fcb-find-identifiers"><span class="std std-ref">How to generate globally unique, resolvable and persistent identifiers</span></a></p></li>
<li><p><a class="reference internal" href="identifier-mapping.html#fcb-interop-idmapping"><span class="std std-ref">How to interlink data from different sources?</span></a></p></li>
</ul>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<hr class="docutils" />
<span class="target" id="id4"></span></div>
<hr class="docutils" />
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Name</p></th>
<th class="text-align:left head"><p>Affiliation</p></th>
<th class="text-align:left head"><p>orcid</p></th>
<th class="text-align:left head"><p>Credit role</p></th>
<th class="text-align:left head"><p>specific contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>Lucas Giovanni Uberti-Bona Marin</p></td>
<td class="text-align:left"><p>Maastricht University</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://orcid.org/0000-0001-6518-9535">0000-0001-6518-9535</a></p></td>
<td class="text-align:left"><p>Writing - Original Draft</p></td>
<td class="text-align:left"><p>Original format</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<p>This page is released under the Creative Commons 4.0 BY license.</p>
<p><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://mirrors.creativecommons.org/presskit/buttons/80x15/png/by-sa.png" height="20"/></a></p>
<div class="toctree-wrapper compound">
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="identifier-mapping.html" title="previous page">How to interlink data from different sources?</a>
    <a class='right-next' id="next-link" href="bridgedb/bridgedb-example.html" title="next page">Identifier mapping using BridgeDb WS</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The FAIRplus project members<br/>
        
            &copy; Copyright 2020 (CC-BY 4.0).<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="row_footer"> This project has received funding from the <a href="https://www.imi.europa.eu/">Innovative Medicines Initiative</a> 2 Joint Undertaking under grant agreement No 802750. This Joint Undertaking receives support from the European Union’s Horizon 2020 research and innovation programme and EFPIA Companies. This communication reflects the views of the authors and neither IMI nor the European Union, EFPIA or any Associated Partners are liable for any use that may be made of the information contained herein. </div>
<div class="row_footer"> This page was created by <a href="https://github.com/jupyter/jupyter-book/graphs/contributors">The Jupyter Book Community</a> </div>
<div class="row_footer"> <div class="column" style="center"><a href="https://europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/eu-flag-recolored-fairplus.png" height="20" alt="EU flag" style="float:center""></a></div> <div class="column"><a href="https://www.imi.europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/imi-logo-fairplus-v1.png" height="20" alt="IMI logo" style="float:center""></a></div> <div class="column"><a href="https://efpia.eu"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/efpia-logo-fairplus2.png" height="20" alt="EFPIA logo" style="float:center""></a></div> <br/> </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>