
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3. Introduction to terminologies and ontologies</title>
    
  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/mermaid.js"></script>
    <script src="../../../_static/mermaid.core.js"></script>
    <script src="../../../_static/mermaid.min.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ontology-why-using-them.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="4. Selecting terminologies and ontologies" href="selecting-ontologies.html" />
    <link rel="prev" title="Using BridgeDb web services" href="bridgedb/bridgedb-example.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ontology-why-using-them.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Introduction to terminologies and ontologies" />
<meta property="og:description" content="Introduction to terminologies and ontologies  &lt;br/&gt;     &lt;script type=&#34;application/ld+json&#34;&gt; {&#34;@context&#34;: {&#34;sdo&#34;: &#34;https://schema.org/&#34;, &#34;bs&#34;: &#34;https://bioschema" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../index.html">
  
  <img src="../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Who should read it
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  RECIPES
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. InChI and SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     3. Minting identifiers with Minid
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     4. Depositing in Zenodo generic repository
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     5. Search engine optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       5.8.1. Data page markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       5.8.2. Dataset page markup with Schema.org
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Data catalogue markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     1. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="bridgedb-recipe.html">
     2. Identifiers mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       Using BridgeDB web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3. Introduction to terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     4. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     5. Requesting new terms
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-operations-tools.html">
     6. Ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="ontology-robot-recipe.html">
     7. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot/MSIO-robot-build-process.html">
       Application ontology for metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="creating_data_dictionary.html">
     8. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     9. Creating a metadata profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="transcriptomics-metadata.html">
       Metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       Metadata profile validation in RDF
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="from-proprietary-to-open-standard-mzml-exemplar.html">
     10. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     11. File format validation, FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ETL-tools.html">
     12. Extraction, transformation, and loading process
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Software licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Data licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data's permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Provenance information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     2. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     4. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Portals and look-up services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selecting a look-up service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. Deploying the EBI OLS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. with FAIREvaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     1. Making omics data matrices FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox.html">
     2. eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb.html">
     3. nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     4. Oncotrack - clinical cohort datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/resolute.html">
     5. RESOLUTE - transcriptomics datasets
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  ABOUT
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/people.html">
     1. Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/code_of_conduct.html">
     2. Code of conduct
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../boilerplate.html">
   Platform
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/the-turing-way.html">
     1. Leveraging the Turing Way Book
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../_sources/content/recipes/interoperability/ontology-why-using-them.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/ontology-why-using-them.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/interoperability/ontology-why-using-them.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   Main objectives
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#controlled-terminology-or-ontology-what-s-the-difference">
   Controlled terminology or ontology: what’s the difference?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#how-are-they-built-and-maintained-and-why-does-it-matter">
   How are they built and maintained and why does it matter?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#why-are-they-useful">
   Why are they useful?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#are-all-ontologies-compatible-with-each-other">
   Are all ontologies compatible with each other?
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#use-cases-and-iterative-approach">
     Use cases and iterative approach
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#selection-criteria">
     Selection criteria
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusions">
   Conclusions
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     What to read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   License
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="introduction-to-terminologies-and-ontologies">
<span id="fcb-interop-whyonto"></span><h1><span class="section-number">3. </span>Introduction to terminologies and ontologies<a class="headerlink" href="#introduction-to-terminologies-and-ontologies" title="Permalink to this headline">¶</a></h1>
<br/>
<hr class="docutils" />
<script type="application/ld+json"> {"@context": {"sdo": "https://schema.org/", "bs": "https://bioschema.org/"}, "@type": ["sdo:HowTo", "bs:TrainingMaterial"], "name": "Introduction to terminologies and ontologies", "audience": ["Data Curator", "Data Manager", "Data Scientist"], "isPartOf": [{"@type": "CreativeWork", "url": "https://fairplus.github.io/the-fair-cookbook/", "name": "The FAIR Cookbook"}], "learningResourceType": "Survey / Review", "identifier": {"@type": "PropertyValue", "name": "FCB019", "url": "https://w3id.org/faircookbook/FCB019"}, "license": {"@type": "CreativeWork", "url": "https://creativecommons.org/licenses/by/4.0/", "name": "CC-BY-4.0"}, "timeRequired": "PT0H15M0S", "totalTime": "PT0H15M0S"} </script>
<div class="card recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
15 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
No
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Introduction to terminologies and ontologies</div>
<img src="/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Survey / Review</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue">Data Curator, Data Manager, Data Scientist</div>
</div>
</div>
<div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB019" class="text--purple-dark">FCB019
</a></div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="main-objectives">
<h2>Main objectives<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¶</a></h2>
<p>The aim of this recipe is to provide a compact introduction about <code class="docutils literal notranslate"><span class="pre">controlled</span> <span class="pre">terminologies</span></code> and <code class="docutils literal notranslate"><span class="pre">ontologies</span></code>, why these resources are central to the preservation of knowledge and data mining and how such resources are developed.</p>
</div>
<div class="section" id="controlled-terminology-or-ontology-what-s-the-difference">
<h2>Controlled terminology or ontology: what’s the difference?<a class="headerlink" href="#controlled-terminology-or-ontology-what-s-the-difference" title="Permalink to this headline">¶</a></h2>
<p>The need for <code class="docutils literal notranslate"><span class="pre">controlled</span> <span class="pre">vocabulary</span></code> often arises in situations where validation of textual information is necessary for operational requirements.
The main initial driver for data entry harmonization is to increase query recall. In its most basic form, <code class="docutils literal notranslate"><span class="pre">keywords</span></code> may be used to perform indexation. However, if relying on user input alone, the chances of typographic errors increases with the number of users. These unavoidable events accumulate over time and end up  hurting the accuracy of search results and this is the reason for offering sets of predefined values. It reduces the noise.
However this can come at the cost of precision, as the predefined terms may not cover the exact thing users may need to describe. Furthermore, term mis-selection by the user is not eliminated and introduces another type of error.</p>
<p>A <code class="docutils literal notranslate"><span class="pre">controlled</span> <span class="pre">terminology</span></code> is a <em><code class="docutils literal notranslate"><span class="pre">normative</span></code></em> collection of terms, the spelling of which is fixed and for which additional information may be provided such as a <code class="docutils literal notranslate"><span class="pre">definition</span></code>, a set of <code class="docutils literal notranslate"><span class="pre">synonyms</span></code>, an <code class="docutils literal notranslate"><span class="pre">editor</span></code>, a <code class="docutils literal notranslate"><span class="pre">version</span></code>, as well as a <code class="docutils literal notranslate"><span class="pre">license</span></code> determining the condition of use. The set of information about a specific controlled terminology term is designated as <code class="docutils literal notranslate"><span class="pre">term</span> <span class="pre">metadata</span></code>. In a controlled terminology, terms appear as a <code class="docutils literal notranslate"><span class="pre">flat</span> <span class="pre">list</span></code>, meaning that no relationship between any of the entities the controlled terminology represents is captured in any formal way.
This is the main drawback and limitation of <code class="docutils literal notranslate"><span class="pre">controlled</span> <span class="pre">terminologies</span></code>, which are often developed to support a data model or an application.</p>
<p>An <code class="docutils literal notranslate"><span class="pre">ontology</span></code> on the other hand, is a <code class="docutils literal notranslate"><span class="pre">formal</span> <span class="pre">representation</span> <span class="pre">of</span> <span class="pre">a</span> <span class="pre">domain</span> <span class="pre">knowledge</span> <span class="pre">where</span> <span class="pre">concepts</span> <span class="pre">are</span> <span class="pre">organized</span> <span class="pre">hierarchically</span></code>. The qualifier <code class="docutils literal notranslate"><span class="pre">formal</span></code> refers to a set of axioms and rules based on logic (e.g. <code class="docutils literal notranslate"><span class="pre">first</span> <span class="pre">order</span> <span class="pre">logic</span></code>) to structure, organize and check the consistency of the term hierarchy. As one can sense right away, ontologies are often a more sophisticated artefact, supported by more advanced theoretical frameworks and dedicated tools to develop them (e.g. Protégé, TopBraid Composer, OBO foundry INCAtools or Robot tool).</p>
</div>
<div class="section" id="how-are-they-built-and-maintained-and-why-does-it-matter">
<h2>How are they built and maintained and why does it matter?<a class="headerlink" href="#how-are-they-built-and-maintained-and-why-does-it-matter" title="Permalink to this headline">¶</a></h2>
<p>In order to improve over simple <code class="docutils literal notranslate"><span class="pre">controlled</span> <span class="pre">terminologies</span></code>, a huge area of research has developed to provide <code class="docutils literal notranslate"><span class="pre">tools</span></code> and <code class="docutils literal notranslate"><span class="pre">frameworks</span></code> supporting the representations of relationships between entities. The field is known as <code class="docutils literal notranslate"><span class="pre">formal</span> <span class="pre">semantics</span></code> in knowledge representation circles. One of the most immediately available examples of <code class="docutils literal notranslate"><span class="pre">entity</span> <span class="pre">relationships</span></code> found in ontologies, and their potential for improving searches, is the <code class="docutils literal notranslate"><span class="pre">is_a</span></code> relationship, which aims to cover the Parent/Child relationship that holds between two entities. For instance:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">Vertebrate</span>
<span class="o">--</span><span class="n">Mammal</span>
<span class="o">---</span><span class="n">Dolphin</span>
<span class="o">--</span><span class="n">Bird</span>
<span class="o">---</span><span class="n">Pigeon</span>
</pre></div>
</div>
<p>In this representation, <code class="docutils literal notranslate"><span class="pre">classes</span></code> are <em>directly</em> asserted (placed) under a parent class if and only if the rule <code class="docutils literal notranslate"><span class="pre">new</span> <span class="pre">class</span> <span class="pre">is</span> <span class="pre">a</span> <span class="pre">child</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">parent</span> <span class="pre">Class</span></code> holds. ‘Orchid’, which in this hierarchy, would not be nested under ‘Vertebrate’.</p>
<p>While working on small structured vocabularies, it is still possible to detect potential errors but this approach does not scale to support real life semantic artefacts which support complex biological and biomedical information systems.
Languages such as <a class="reference external" href="https://www.w3.org/TR/2014/NOTE-rdf11-primer-20140624/">RDF</a>, <a class="reference external" href="https://www.w3.org/2004/02/skos/">SKOS</a>, and <a class="reference external" href="https://www.w3.org/OWL/">OWL</a>, exist to provide the expressivity required to establish relations between entities.
In turn, building on these formal rules, automatic classifiers, known as a <code class="docutils literal notranslate"><span class="pre">reasoner</span></code>, can inspect semantic artefacts to detect inconsistencies and suggest parent classes.
This is a step known as <code class="docutils literal notranslate"><span class="pre">inference</span></code>, where new knowledge is produced by the software agent rather than direct assertion by humans.
This provides a significant support, even if far from supporting all the subtleties of actual knowledge.</p>
<p>There are <code class="docutils literal notranslate"><span class="pre">six</span> <span class="pre">important</span> <span class="pre">features</span></code> to consider when selecting a semantic artefact for making FAIR datasets:</p>
<ol class="simple">
<li><p><strong>What license and terms of use does it mandate?</strong></p></li>
<li><p><strong>What format does it come in?</strong></p></li>
<li><p><strong>Is it well maintained, i.e. frequent release, term requests handling, versioning and deprecation policies clarified?</strong></p></li>
<li><p><strong>Are there stable persistent resolvable identifiers for all terms?</strong></p></li>
<li><p><strong>Who use it and what resources are being annotated with it?</strong></p></li>
<li><p><strong>Is it well documented?  There should be enough metadata for each class in the artefact and enough metadata about the artefact itself.</strong></p></li>
</ol>
</div>
<div class="section" id="why-are-they-useful">
<h2>Why are they useful?<a class="headerlink" href="#why-are-they-useful" title="Permalink to this headline">¶</a></h2>
<p>As outlined in the introduction, the most immediate use for a controlled terminology is to ensure consistency in data entry. Controlled terminologies are important tools to improve data indexing and query recall.
The usefulness of ontologies and controlled vocabularies goes beyond this initial use.
The main purpose of biomedical ontologies is to structure knowledge so that it can be operated on by software agents.</p>
<p>One needs to also understand that the two processes coexist and operate in parallel.
As more experiments are performed, new discoveries are made.
This new knowledge needs to be represented in the domain ontology so the new notions can be used to annotate the results of earlier experiments in the context of retrospective analysis.</p>
<p>For example, he <a class="reference external" href="http://www.obofoundry.org/ontology/go.html">Gene Ontology (GO)</a> is a widely used resource to describe <code class="docutils literal notranslate"><span class="pre">Molecular</span> <span class="pre">Processes</span></code>, <code class="docutils literal notranslate"><span class="pre">Biological</span> <span class="pre">Functions</span></code> and <code class="docutils literal notranslate"><span class="pre">Molecular</span> <span class="pre">Components</span></code>.
The <a class="reference external" href="http://geneontology.org/docs/go-consortium/">Gene Ontology Consortium</a> maintains the controlled vocabulary and also releases of Genome Wide <a class="reference external" href="http://geneontology.org/docs/go-annotations/">Gene Ontology Annotations</a>.
These are resources which associate genes and genomic features found in those genomes with GO terms. These are very important resources especially in the context of genome wide analysis such as transcriptomics profiling analysis.</p>
<p>A particular type of analysis, <a class="reference external" href="http://geneontology.org/docs/go-enrichment-analysis/"><code class="docutils literal notranslate"><span class="pre">enrichment</span> <span class="pre">analysis</span></code></a>, relies on the availability of such annotations to detect departures from the expected probability distribution in an expression profile and which biological processes are most affected in specific conditions.</p>
<p>The applications are plentiful. The importance of ontologies for structuring information will only grow with the need to obtain Machine Learning ready datasets and speed up the readiness of datasets. This is what FAIR is all about.</p>
<p>So ontologies are of particularly help for the following tasks:</p>
<ul class="simple">
<li><p><strong>Improve query recall:</strong> Given a ‘search string’, having a resource which holds synonyms can be used by a search index to retrieve data annotated with a synonym.</p></li>
<li><p><strong>Enable query expansion:</strong> Owing to the hierarchical (parent/child) structure of ontologies, a search index exploiting this information can retrieve all datasets annotated with a child term of items matching the input search string. For example, searching with the string “breast cancer” againts an ontology aware search index could return records annotated with <code class="docutils literal notranslate"><span class="pre">Paget's</span> <span class="pre">Disease</span></code> or <code class="docutils literal notranslate"><span class="pre">ductal</span> <span class="pre">carcinoma</span> <span class="pre">in-situ</span> <span class="pre">(DCIS)</span></code>, both of which are types of mammary gland malignancies.</p></li>
<li><p><strong>Build knowledge graphs:</strong> Ontology languages can be used to represent domain knowledge and build reference terminologies but the same technologies constitute powerful tools for modelling instance datasets as nodes in a graph and linking resources together.</p></li>
</ul>
</div>
<div class="section" id="are-all-ontologies-compatible-with-each-other">
<h2>Are all ontologies compatible with each other?<a class="headerlink" href="#are-all-ontologies-compatible-with-each-other" title="Permalink to this headline">¶</a></h2>
<p>There is not a simple answer to that question as it depends heavily on the type of tasks data scientists have in mind.
If the purpose is simply to improve query recall on a limited set of fields, a curation policy could be devised to mix and match resources to meet the needs at hands, possibly by building an application ontology, i.e. an ontology specifically for designed for the use case with terms drawn from existing ontologies.</p>
<p>However, in a more integrated framework, it is important to be aware of some of the development choices made by the maintainers of the semantic artefacts.</p>
<ul class="simple">
<li><p><strong>In the context of basic research and model organism based research</strong>, the <strong><code class="docutils literal notranslate"><span class="pre">OBO</span> <span class="pre">foundry</span></code></strong> is an organization which coordinates the development of interoperable resources. GO, mentioned earlier is one of them. The establishment of domain specific reference ontologies sharing the same underlying rules means that some level of compositional development can be done. This means that axioms can be built connecting classes from compatible resources.
This point becomes particularly important when considering the role of the <code class="docutils literal notranslate"><span class="pre">reasoner</span></code> when assessing and checking the consistency of artefacts themselves but also when analysing instance datasets themselves.</p></li>
<li><p><strong>In the context of observation studies</strong>, the Observational Medical Outcomes Partnership (<a class="reference external" href="https://ohdsi.org/omop/">OMOP</a>) model also relies on controled terminologies such as <a class="reference external" href="https://www.snomed.org/snomed-ct/">SNOMED-CT</a>, <a class="reference external" href="https://www.nlm.nih.gov/research/umls/rxnorm/index.html">RxNORM</a> for drugs and <a class="reference external" href="https://loinc.org/">LOINC</a> for clinical and laboratory test descriptions.</p></li>
<li><p><strong>In the context of Clinical Data collections</strong>, the Clinical Data Interchange Standards Consortium (<a class="reference external" href="https://www.cdisc.org/">CDISC</a>) models work tightly with <a class="reference external" href="https://www.cdisc.org/standards/terminology/controlled-terminology">CDISC Terminology</a>, National Cancer Institute’s Enterprise Vocabulary Services (<a class="reference external" href="https://evs.nci.nih.gov/">EVS</a>) and also recommend use of <a class="reference external" href="https://www.snomed.org/snomed-ct/">SNOMED-CT</a> and terminologies such as <a class="reference external" href="https://loinc.org/">LOINC</a>, both of which come with specific licensing terms users need to be familiar with.</p></li>
</ul>
<div class="section" id="use-cases-and-iterative-approach">
<h3>Use cases and iterative approach<a class="headerlink" href="#use-cases-and-iterative-approach" title="Permalink to this headline">¶</a></h3>
<p>The use and implementation of common terminologies will enable a normalization/harmonization of variable labels (data label) and allowed values (data term) when querying a database. Implementing the use of common terminologies in the curation workflow will ensure consistency of the annotation across all studies.</p>
</div>
<div class="section" id="selection-criteria">
<h3>Selection criteria<a class="headerlink" href="#selection-criteria" title="Permalink to this headline">¶</a></h3>
<p>A set of widely accepted criteria for selecting terminologies (or other reporting standards) do not exists. However, the initial work by the Clinical and Translational Science Awards’ (CTSA) Omics Data Standards Working Group and FAIRSharing (<a class="reference external" href="http://jamia.bmj.com/content/early/2013/10/03/amiajnl-2013-002066.long">http://jamia.bmj.com/content/early/2013/10/03/amiajnl-2013-002066.long</a>) has been used as starting point to define possible criteria for excluding and/or including a terminology resource.</p>
<ul class="simple">
<li><p><strong>Exclusion criteria</strong>:</p>
<ul>
<li><p>🔸 Absent licence or terms of use (<em>indicator of usability</em>)</p></li>
<li><p>🔸 Restrictive licences or terms of use with restrictions on redistribution and reuse</p></li>
<li><p>🔸 Absence of term definitions</p></li>
<li><p>🔸 Absence of sufficient class metadata (<em>indicator of quality</em>)</p></li>
<li><p>🔸 Absence of sustainability indicators (<em>absence of funding records</em>)</p></li>
</ul>
</li>
<li><p><strong>Inclusion criteria</strong>:</p>
<ul>
<li><p>🔰  Scope and coverage meets the requirements of the concept identified</p></li>
<li><p>🔰  Unique URI, textual definition and IDs for each term</p></li>
<li><p>🔰  Resource releases are versioned</p></li>
<li><p>🔰  Size of resource (<em>indicator of coverage</em>)</p></li>
<li><p>🔰  Number of classes and subclasses (<em>indicator of depth</em>)</p></li>
<li><p>🔰  Number of terms having definitions and synonyms (<em>indicator of richness</em>)</p></li>
<li><p>🔰  Presence of a help desk and contact point (<em>indicator of community support</em>)</p></li>
<li><p>🔰  Presence of term submission tracker/issue tracker (<em>indicator of resource agility and capability to grow upon request</em>)</p></li>
<li><p>🔰  Potential integrative nature of the resource (<em>as indicator of translational application potential</em>)</p></li>
<li><p>🔰  Licensing information available (<em>as indicator of freedom to use</em>)</p></li>
<li><p>🔰  Use of a top level ontology (<em>as indicator of a resource built for generic use</em>)</p></li>
<li><p>🔰  Pragmatism (<em>as indicator of actual, current real life practice)</em></p></li>
<li><p>🔰  Possibility of collaborating: the resource accepts complaints/remarks that aim to fix or improve the terminology, while the resource organisation commits to fix or improve the terminology in brief delays (one month after receipt?)</p></li>
</ul>
</li>
</ul>
<p>These criteria are simply indicative and need to be modulated depending on the <code class="docutils literal notranslate"><span class="pre">contexts</span></code> described in the introduction, as specific constraints (e.g. regulatory requirements) may take precedence over some of the criteria listed here.</p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="conclusions">
<h2>Conclusions<a class="headerlink" href="#conclusions" title="Permalink to this headline">¶</a></h2>
<p>Choosing an ontology and semantic resources is a complex issue, which requires careful consideration, taking into account the research context of the data production workflow and regulatory requirements that may apply. The choices made affect the integrative potential of a dataset as they influence the level of <code class="docutils literal notranslate"><span class="pre">interoperability</span></code>.
Clearly, declaring the semantic resources used to annotate a dataset also influences the <code class="docutils literal notranslate"><span class="pre">findability</span></code> and <code class="docutils literal notranslate"><span class="pre">reusability</span></code> and it is good practice to do so as it allows potential users to gauge the amount of mapping work that may be required to combine two datasets.</p>
<div class="section" id="what-to-read-next">
<h3>What to read next?<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="selecting-ontologies.html#fcb-interop-selectonto"><span class="std std-ref">Selecting terminologies and ontologies</span></a></p></li>
<li><p><a class="reference internal" href="ontology-new-term-request-recipe.html#fcb-interop-ontorequest"><span class="std std-ref">Requesting new terms</span></a></p></li>
<li><p><a class="reference internal" href="ontology-robot-recipe.html#fcb-interop-ontorobot"><span class="std std-ref">Building an application ontology with ROBOT</span></a></p></li>
<li><p><a class="reference internal" href="ontology-operations-tools.html#fcb-interop-onto-op"><span class="std std-ref">Ontology-related tools and services</span></a></p></li>
</ul>
</div>
<!-- > * TODO: {ref}`fcb-find-biosolr` -->
</div>
<hr class="docutils" />
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<ol class="simple">
<li><p>RDF. <a class="reference external" href="https://www.w3.org/TR/2014/NOTE-rdf11-primer-20140624/">https://www.w3.org/TR/2014/NOTE-rdf11-primer-20140624/</a></p></li>
<li><p>SKOS. <a class="reference external" href="https://www.w3.org/2004/02/skos/">https://www.w3.org/2004/02/skos/</a></p></li>
<li><p>OWL. <a class="reference external" href="https://www.w3.org/OWL/">https://www.w3.org/OWL/</a></p></li>
<li><p>Hermit. <a class="reference external" href="http://www.hermit-reasoner.com/">http://www.hermit-reasoner.com/</a></p></li>
<li><p>Elk. <a class="reference external" href="http://www.cs.ox.ac.uk/isg/tools/ELK/">http://www.cs.ox.ac.uk/isg/tools/ELK/</a></p></li>
<li><p>OBO Foundry. <a class="reference external" href="http://obofoundry.org/">http://obofoundry.org/</a></p></li>
<li><p>CDISC. <a class="reference external" href="https://www.cdisc.org/standards">https://www.cdisc.org/standards</a></p></li>
<li><p>CDISC Controlled Terminology. <a class="reference external" href="https://www.cdisc.org/standards/terminology">https://www.cdisc.org/standards/terminology</a></p></li>
<li><p>LOINC. <a class="reference external" href="https://loinc.org/">https://loinc.org/</a></p></li>
<li><p>Gene Ontology. <a class="reference external" href="http://geneontology.org/">http://geneontology.org/</a></p></li>
<li><p>Protégé. <a class="reference external" href="https://protege.stanford.edu/">https://protege.stanford.edu/</a></p></li>
<li><p>Topbraid composer. <a class="reference external" href="https://www.topquadrant.com/products/topbraid-composer/">https://www.topquadrant.com/products/topbraid-composer/</a></p></li>
<li><p>INCAtools. <a class="reference external" href="https://github.com/INCATools">https://github.com/INCATools</a></p></li>
<li><p>ROBOT. <a class="reference external" href="https://doi.org/10.1186/s12859-019-3002-3">R.C. Jackson, J.P. Balhoff, E. Douglass, N.L. Harris, C.J. Mungall, and J.A. Overton. ROBOT: A tool for automating ontology workflows. BMC Bioinformatics, vol. 20, July 2019.</a></p></li>
</ol>
</div>
<hr class="docutils" />
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Orcid</p></th>
<th class="head"><p>Affiliation</p></th>
<th class="head"><p>Type</p></th>
<th class="text-align:center head"><p>Elixir Node</p></th>
<th class="text-align:center head"><p>Credit Role</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><div class="firstCol"><a target="_blank" href='https://github.com/proccaserra'><img class='avatar-style' src='https://avatars.githubusercontent.com/proccaserra'></img><div class="d-block">Philippe Rocca-Serra</div></a>  </div></p></td>
<td><p><a target="_blank" href='https://orcid.org/0000-0001-9853-5668'><i class='fab fa-orcid fa-2x text--orange'></i></a></p></td>
<td><p>University of Oxford</p></td>
<td><p><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td class="text-align:center"><p><img class='elixir-style' src='/the-fair-cookbook/_static/images/logo/Elixir/ELIXIR-UK.svg' ></img></p></td>
<td class="text-align:center"><p>Writing - Original draft</p></td>
</tr>
<tr class="row-odd"><td><p><div class="firstCol"><a target="_blank" href='https://github.com/alasdairgray'><img class='avatar-style' src='https://avatars.githubusercontent.com/alasdairgray'></img><div class="d-block">Alasdair J G Gray</div></a>  </div></p></td>
<td><p><a target="_blank" href='http://orcid.org/0000-0002-5711-4872'><i class='fab fa-orcid fa-2x text--orange'></i></a></p></td>
<td><p>Heriot-Watt University</p></td>
<td><p><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td class="text-align:center"><p><img class='elixir-style' src='/the-fair-cookbook/_static/images/logo/Elixir/ELIXIR-UK.svg' ></img></p></td>
<td class="text-align:center"><p>Writing - Review &amp; Editing</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<p>This page is released under the Creative Commons 4.0 BY license.</p>
<p><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://mirrors.creativecommons.org/presskit/buttons/80x15/png/by-sa.png" height="20"/></a></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="bridgedb/bridgedb-example.html" title="previous page">Using BridgeDb web services</a>
    <a class='right-next' id="next-link" href="selecting-ontologies.html" title="next page"><span class="section-number">4. </span>Selecting terminologies and ontologies</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
            &copy; Copyright 2020.<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="container-lg mt-3"> <div class="row"> <div class="col-xl-3"> <div class="footerImages"> <a href="https://europa.eu/"><img src="/the-fair-cookbook/_static/images/logo/eu-flag.png" height="20" alt="EU flag" style="float:center""></a> <a href="https://www.imi.europa.eu/"><img src="/the-fair-cookbook/_static/images/logo/imi-logo.png" height="20" alt="IMI logo" style="float:center""></a> <a href="https://efpia.eu"><img src="/the-fair-cookbook/_static/images/logo/efpia-logo.png" height="20" alt="EFPIA logo" style="float:center""></a> </div> <div class="mb-3"> Grant agreement 802750 </div> </div> <div class="col-xl-6 d-flex"> <div class="container"> <div class="row"> <div class="col-6 d-flex" style="justify-content:center"> <ul> <li><a href="mailto:fairplus-cookbook@elixir-europe.org"> Contact us </a></li> <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li> </ul> </div> <div class="col-6 d-flex" style="justify-content:center"> <ul> <li> <a href="/the-fair-cookbook/content/recipes/boilerplate/people.html"> Contributors </a> </li> </li> <li> <a href="/the-fair-cookbook/content/recipes/help.html"> How to contribute </a> </li> </ul> </div> </div> </div> </div> <div class="col-xl-3"> <a href="https://elixir-europe.org/" target="_blank"><img src="/the-fair-cookbook/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo" style="float:center""></a> <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a> </div> </div> </div> <div class="row_footer my-3"> The FAIR Cookbook is licensed under the <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">Creative Commons Attribution 4.0 International</a> (CC BY 4.0) license. </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>